### PR TITLE
Added m2m for licence in `Assign licence` transition.

### PR DESCRIPTION
**Pull request by ghost**
Before:
User could pick only one licence in `Assign licence` tranision.
Now:
User can pick many licences in `Assign licence` tranision.

### CODE REVIEW

File: src/ralph/back_office/models.py [hunk 1]
```diff
@@ -229,10 +231,10 @@ def assign_user(cls, instances, request, **kwargs):
     )
     def assign_licence(cls, instances, request, **kwargs):
         for instance in instances:
-            BaseObjectLicence.objects.get_or_create(
-                base_object=instance,
-                licence=Licence.objects.get(pk=kwargs['licence'])
-            )
+            for pk in kwargs['licence']:
+                BaseObjectLicence.objects.get_or_create(
+                    base_object=instance, licence=Licence.objects.get(pk=pk)
```
Comment (mkurek) [inline src/ralph/back_office/models.py]: why not `base_object=instance, licence_id=pk`?

File: src/ralph/lib/transitions/models.py [hunk 2]
```diff
@@ -101,8 +101,11 @@ def _get_history_dict(data, instance, runned_funcs):
                     value = str(field.rel.to.objects.get(pk=v))
             except FieldDoesNotExist:
                 field = func.form_fields[k]['field']
-                if isinstance(field, forms.ChoiceField):
+                if type(field) == forms.ChoiceField:
```
Comment (mkurek) [inline src/ralph/lib/transitions/models.py]: i'd prefer not to use `type` comparison. you could swap these if-else to first check `if isinstance(field, forms.ModelMultipleChoiceField):`, then `elif isinstance(field, forms.ChoiceField):`

File: src/ralph/lib/transitions/views.py [hunk 3]
```diff
@@ -62,10 +62,15 @@ def form_fields_from_actions(self):
                     field = model._meta.get_field(
                         options['autocomplete_field']
                     )
+                    widget_options = {}
+                    autocomplete_name = options.get('autocomplete_field', None)
+                    if autocomplete_name in ['licence']:
```
Comment (mkurek) [inline src/ralph/lib/transitions/views.py]: http://www.nooooooooooooooo.com/
please add this as another parameter to transitions fields definition or deduct if from field type

File: src/ralph/back_office/models.py [hunk 4]
```diff
@@ -221,18 +221,21 @@ def assign_user(cls, instances, request, **kwargs):
     @transition_action(
         form_fields={
             'licence': {
```
Comment (mkurek) [inline src/ralph/back_office/models.py]: licences?

File: src/ralph/back_office/tests/test_models.py [hunk 5]
```diff
@@ -296,6 +297,20 @@ def test_convert_to_data_center_asset(self):
...
+    def test_assign_many_licences(self):
+        asset = BackOfficeAssetFactory()
+        licences = [LicenceFactory() for i in range(2)]
+        self.assertFalse(asset.licences.all())
+
+        BackOfficeAsset.assign_licence(
+            instances=[asset], request=None, licence=licences,
+        )
+
+        self.assertEqual(
```
Comment (mkurek) [inline src/ralph/back_office/tests/test_models.py]: I'd use https://docs.python.org/3.4/library/unittest.html#unittest.TestCase.assertCountEqual here

File: src/ralph/lib/transitions/views.py [hunk 6]
```diff
@@ -62,10 +62,12 @@ def form_fields_from_actions(self):
                     field = model._meta.get_field(
                         options['autocomplete_field']
                     )
+                    widget_options = options.get('widget_options', {})
                     options['field'].widget = AutocompleteWidget(
                         field=field,
                         admin_site=ralph_site,
                         request=self.request,
+                        **widget_options
```
Comment (mkurek) [inline src/ralph/lib/transitions/views.py]: why not just `**options.get('widget_options', {})`?

**Comment by szok** [comment]
:ok:

**Comment by mkurek** [comment]
:ok:

**Pull request merged**